Metalama.Framework.Engine/Utilities/Roslyn/SymbolExtensions.cs

```diff
@@ -170,15 +171,47 @@ internal static bool IsAccessor( this IMethodSymbol method )
                 r => r.GetSyntax() is MemberDeclarationSyntax member && member.Modifiers.Any( m => m.IsKind( kind ) ) );
         }
 
-        internal static bool IsInterfaceMemberImplementation( this ISymbol symbol )
+        private static ImmutableArray<ISymbol> ExplicitInterfaceImplementations( this ISymbol symbol )
             => symbol switch
             {
-                IMethodSymbol methodSymbol => methodSymbol.ExplicitInterfaceImplementations.Any(),
-                IPropertySymbol propertySymbol => propertySymbol.ExplicitInterfaceImplementations.Any(),
-                IEventSymbol eventSymbol => eventSymbol.ExplicitInterfaceImplementations.Any(),
-                _ => false
+                IEventSymbol @event => ImmutableArray<ISymbol>.CastUp( @event.ExplicitInterfaceImplementations ),
```

Comment:
This refactoring make memory allocation worse in case that IsExplicitInterfaceMemberImplementation is used. I would keep IsExplicitInterfaceMemberImplementation as a different implementation that would be more optimized.

Reply:
`CastUp` doesn't allocate:

> Initializes a new instance of the `ImmutableArray<T>` struct based on the contents of an existing instance, allowing a covariant static cast to efficiently reuse the existing array.